tests/server/activity-event.js

```diff
@@ -0,0 +1,154 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+define([
+  'intern!object',
+  'intern/chai!assert',
+  'intern/dojo/node!../../server/lib/activity-event',
+  'intern/dojo/node!bluebird',
+  'intern/dojo/node!path',
+  'intern/dojo/node!sinon',
+], function (registerSuite, assert, activityEvent, Promise, path, sinon) {
+  var write;
+
+  registerSuite({
+    name: 'activity-event',
+
+    'interface is correct': function () {
+      assert.isFunction(activityEvent);
+      assert.lengthOf(activityEvent, 3);
+    },
+
+    'call activityEvent with service, without client_id': {
+      setup: function () {
+        write = process.stderr.write;
+        process.stderr.write = sinon.spy();
+        activityEvent('mock event', { a: 'b', c: 'd' }, {
+          headers: { 'user-agent': 'foo' },
+          originalUrl: 'bar',
+          query: {
+            context: 'mock context',
+            entrypoint: 'mock entrypoint',
```

Comment:
Little remembered fact - old versions of Sync opened FxA with [an `entryPoint` query parameter](https://github.com/mozilla/fxa-content-server/blob/5a76e3b5632738b44efb14d40f28ee2d6acea75a/app/scripts/models/reliers/relier.js#L85). Do you care? 

It doesn't look [like that discrepancy exists anymore](https://dxr.mozilla.org/mozilla-central/search?q=entryPoint%3A&redirect=false&case=true)